components/split-view/splitViewList.jsx

```diff
@@ -0,0 +1,276 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+/* eslint-disable class-methods-use-this */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { SPLIT_VIEW_LIST } from '../../utilities/constants';
+
+import Icon from '../icon';
+import SplitViewListItemContent from './private/list-item-content';
+import listItemWithContent from './private/list-item-with-Content';
+
+export const SORT_OPTIONS = Object.freeze({
+	UP: 'up',
+	DOWN: 'down'
+});
+
+const propTypes = {
+	/**
+	 * **Assistive text for accessibility**
+	 * * `list`: aria label for the list
+	 * * `sort`
+	 * * * `sortedBy`: Clickable sort header for the list.
+	 * * * `descending`: Descending sorting.
+	 * * * `ascending`: Ascending sorting.
+	 */
+	assistiveText: PropTypes.shape({
+		list: PropTypes.string,
+		sort: PropTypes.shape({
+			sortedBy: PropTypes.string,
+			descending: PropTypes.string,
+			ascending: PropTypes.string
+		}),
+		unreadItem: PropTypes.string
+	}),
+	/**
+	 * CSS classes to be added to the parent `div` tag.
+	 */
+	className: PropTypes.oneOfType([PropTypes.array, PropTypes.object, PropTypes.string]),
+	/**
+	 * Event Callbacks
+	 * * `onSelect`: Called when a list item is selected.
+	 * * * @param event {object} List item click event
+	 * * * @param Meta {object}
+	 * * * * selectedItems {array} List of selected items.
+	 * * * * item {object} Last selected item.
+	 * * `onSort`: Called when the list is sorted.
+	 * * * @param event {object} Sort click event
+
+	 */
+	events: PropTypes.shape({
+		onSelect: PropTypes.func.isRequired,
+		onSort: PropTypes.func
+	}),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * **Text labels for internationalization**
+	 * * `header`: This is the header of the list.
+	 */
+	labels: PropTypes.shape({
+		header: PropTypes.string
+	}),
+	/**
+	 * The direction of the sort arrow. Option are:
+	 * * SORT_OPTIONS.UP: `up`
+	 * * SORT_OPTIONS.DOWN: `down`
+	 */
+	sortDirection: PropTypes.oneOf([SORT_OPTIONS.UP, SORT_OPTIONS.DOWN]),
+	/**
+	 * Allows multiple item to be selection
+	 */
+	multiple: PropTypes.bool,
+	/**
+	 * The list of items.
+	 * It is recommended that you have a unique `id` for each item.
```

Comment:
It defaults to the [`index` if no `id`](https://github.com/kevinlambert/design-system-react/blob/0522acaf358082340fad6c8bacf5be7e8c408797/components/split-view/splitViewList.jsx#L247) is supplied so it should be fine. 

But React recommends against using the index in particular for sorting
https://reactjs.org/docs/lists-and-keys.html

> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny’s article for an in-depth explanation on the negative impacts of using an index as a key. If you choose not to assign an explicit key to list items then React will default to using indexes as keys.

Reply:
I'm sorry. I mistyped that. We are in agreement. My concern is that I'd remove `index` so that it _will_ crash if no `id` is present. Otherwise, it might work if devs don't provide an id. We want it to crash if you do it wrong. 😉